gossip/minimal/mysql/hawkstate.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package mysql provides a MySQL based implementation of persistent
+// state management for the goshawk tool.
+package mysql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/google/certificate-transparency-go/gossip/minimal"
+)
+
+func NewStateManager(ctx context.Context, db *sql.DB) (minimal.ScanStateManager, error) {
+	tx, err := db.BeginTx(ctx, nil /* opts */)
```

Comment:
I still think these might benefit from having explicit "restore" APIs but it's purely an opinion.

Reply:
Pulled out into a separate `restore()` method.